Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -2187,15 +2192,13 @@ - (void)proofreadingSession:(WTSession *)session didReceiveSuggestions:(NSArray<
         [_writingToolsTextSuggestions setObject:suggestion forKey:suggestion.uuid];
     }
 
-    _page->proofreadingSessionDidReceiveSuggestions(*webSession, replacementData, *webContext, finished, [webSession, replacementData, webContext, finished, page = WeakPtr { _page.get() }] {
-        if (!page)
-            return;
+    NSInteger delta = [WebKit::getWKIntelligenceTextEffectCoordinatorClass() characterDeltaForReceivedSuggestions:suggestions];
 
-        // FIXME: Wait for the animation to finish before invoking this method.
-        page->proofreadingSessionDidCompletePartialReplacement(*webSession, replacementData, *webContext, finished, [] {
-            // FIXME: Once the animations are implemented, this completion handler will be used to inform the coordinator that this async operation is complete.
+    [_intelligenceTextEffectCoordinator requestReplacementWithProcessedRange:range characterDelta:delta operation:^(void (^completion)(void)) {
```

Comment:
I think you should use a lambda + `makeBlockPtr` here. At the very least `_page` should be weakly referenced.